lib/Operation.php

```diff
@@ -122,4 +134,84 @@ private function getAppDataFolderName() {
 
 		return 'appdata_' . $instanceId;
 	}
+
+	/**
+	 * returns a translated name to be presented in the web interface
+	 *
+	 * Example: "Automated tagging" (en), "Aŭtomata etikedado" (eo)
+	 *
+	 * @since 18.0.0
+	 */
+	public function getDisplayName(): string {
+		return $this->l->t('Automated tagging');
+	}
+
+	/**
+	 * returns a translated, descriptive text to be presented in the web interface.
+	 *
+	 * It should be short and precise.
+	 *
+	 * Example: "Tag based automatic deletion of files after a given time." (en)
+	 *
+	 * @since 18.0.0
+	 */
+	public function getDescription(): string {
+		return $this->l->t('Automated tagging of files');
+	}
+
+	/**
+	 * returns the URL to the icon of the operator for display in the web interface.
+	 *
+	 * Usually, the implementation would utilize the `imagePath()` method of the
+	 * `\OCP\IURLGenerator` instance and simply return its result.
+	 *
+	 * Example implementation: return $this->urlGenerator->imagePath('myApp', 'cat.svg');
+	 *
+	 * @since 18.0.0
+	 */
+	public function getIcon(): string {
+		return $this->urlGenerator->imagePath('files_automatedtagging', 'app.svg');
+	}
+
+	/**
+	 * returns whether the operation can be used in the requested scope.
+	 *
+	 * Scope IDs are defined as constants in OCP\WorkflowEngine\IManager. At
+	 * time of writing these are SCOPE_ADMIN and SCOPE_USER.
+	 *
+	 * For possibly unknown future scopes the recommended behaviour is: if
+	 * user scope is permitted, the default behaviour should return `true`,
+	 * otherwise `false`.
+	 *
+	 * @since 18.0.0
+	 */
+	public function isAvailableForScope(int $scope): bool {
+		return true;
+	}
+
+	/**
+	 * Is being called by the workflow engine when an event was triggered that
+	 * is configured for this operation. An evaluation whether the event
+	 * qualifies for this operation to run has still to be done by the
+	 * implementor.
+	 *
+	 * If the implementor is an IComplexOpe    ration, this method will not be
+	 * called automatically. It can be used or left as no-op by the implementor.
+	 *
+	 * @since 18.0.0
+	 */
+	public function onEvent(string $eventName, GenericEvent $event, IRuleMatcher $matcher): void {
```

Comment:
PHP Fatal error:  Declaration of OCA\FilesAutomatedTagging\Operation::onEvent(string $eventName, Symfony\Component\EventDispatcher\GenericEvent $event, OCP\WorkflowEngine\IRuleMatcher $matcher): void must be compatible with OCP\WorkflowEngine\IOperation::onEvent(string $eventName, OCP\EventDispatcher\Event $event, OCP\WorkflowEngine\IRuleMatcher $ruleMatcher): void in /home/travis/build/nextcloud/server/apps/files_automatedtagging/lib/Operation.php on line 41

Reply:
wasn't committed yet, because it was waiting for a server pr to be merged. fixed now.